### ISSUE TITLE
Document text search result not on page shown.

### ISSUE THREAD

**Issue by Biep** [koreader/koreader#12686]
* KOReader version: many, currently 2027.07-264-g42d3f2503_2024-10-23
* Device: PocketBook Era (P700)

#### Issue
See [here](https://github.com/koreader/koreader/issues/12589#issuecomment-2407006376).
Sometimes when I search text in a document (I **think** only PDFs, especially large ones), going forward or backward with the `>` or `<` buttons on the pop-up in the screen centre, the search stops on a page where no occurrence of the text searched can be found.  Sometimes I find the text on a previous or subsequent page.

#### Steps to reproduce
Open a large PDF document (especially [the Palmer Bible](https://bibletranslation.ws/trans/holybible.pdf)), search for some word that is know to occur, move a few times to the next or previous occurrence.
The issue is intermittent, so this is likely to show no issue at all.

I hope to remember next time I hit upon the issue to retry with precisely the same search.  If I find a search that consistently goes wrong in the way described. I'll add it here.

**Comment by jonnyl2** [comment]
Could it be that your view mode is not showing the entire page at once (e.g., zoom-to-width, manual zoom, reflow, etc.)? This search widget only jumps to the first section of each page with a search result; the search result is likely located outside of view. To make sure you're not missing any search results, switch to a view showing the entire page, or use the 'Search All' feature. This is indeed only an issue with PDF/DJVU documents.

**Comment by Biep** [comment]
I indeed do zoom to width, so that could explain it. 👍️
But shouldn't the search feature adapt to that, and go to the view window that in fact contains the text sought for?
(If that is hard to do, maybe an explanation would be handy, a bit akin to the one for the page browser?)

**Comment by hius07** [comment]
An easy way to fix is to switch to the "Skim mode" when showing search results.

**Comment by jonnyl2** [comment]
@hius07:  It doesn't seem to be possible to switch to skim mode while cycling through the search results. (Unless you were talking about 'Search All'-results, then yes, but the highlighted search terms do indeed disappear.)

**Comment by hius07** [comment]
Not SearchAll. Switch to "page" view, then start searching.
I need to try.

**Comment by Olowojaye** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by hius07** [comment]
Yes, we can switch to the "full page view mode" before start of search, show results in this mode, and then switch back to the user's mode on closing the search dialog.
@Biep to try this approach you can see how the "Skim document" works, it makes the same switching.

**Comment by hius07** [comment]
Also asking @ichnilatis-gr as a heavy pdf reader, is it useful to switch to the "full page view" when searching in the document? Expecially in the reflow mode, some search results can be out of the screen.

**Comment by ichnilatis-gr** [comment]
> Also asking @ichnilatis-gr as a heavy pdf reader, is it useful to switch to the "full page view" when searching in the document? Expecially in the reflow mode, some search results can be out of the screen.

Why not? If the pages are not so big and we can see the words... I haven't used word search much in pdf and I don't particularly use reflow mode.

**Comment by hius07** [comment]
I see the following ways to fix the issue.

("arrow button" is a button in the dialog that popups when the first search results are shown)

(1) On search, switch the view mode to the "full page view" (like we do for SkimTo).
PRO: all found entries are visible, pressing an arrow button shows the next/previous page with found entries.
CONTRA: text may be small and unreadable.

(2) Keep current view mode, show found entries in every visible part of the page.
Pressing an arrow button shows the next/previous **part of the page** with found entries or the next/previous **page** with found entries.
PRO: found entries are readable.
CONTRA: we cannot jump directly to a part of the page with a found entry. We must show **the first** visible part of the page and then press the forward arrow button to show next parts.
To continue searching forward from the current page, we must press the forward arrow button until the last part of the page becomes visible, and then the next pressing starts searching in the following pages.

(3) Have a setting to choose (1) or (2).

**Comment by Biep** [comment]
Can't wait to try the solution!
Thanks all!
